app/src/main/java/com/madonasyombua/growwithgoogleteamproject/database/LoginManager.java

```diff
@@ -0,0 +1,53 @@
+package com.madonasyombua.growwithgoogleteamproject.database;
+
+import android.app.Activity;
+import android.support.annotation.NonNull;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.madonasyombua.growwithgoogleteamproject.models.User;
+
+/**
+ * Created by mahersoua on 23/02/2018.
+ */
+
+public class LoginManager {
+
+    private static final String TAG = "DataManger";
+    private static FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();
+    private static FirebaseUser mCurrentUser;
+
+    public static FirebaseUser registerUser(final Activity activity, User user){
+        firebaseAuth.createUserWithEmailAndPassword(user.getEmail(), user.getPassword())
```

Comment:
I believe this method is asynchronous, not sure if it will return correctly. Has it been tested?

Reply:
Yes of course I tested, it works, I think that async task managed by Firebase sdk

Reply:
Awesome.

Reply:
sorry @adevezin I miss understood your comment, yes it is async I fixed that by adding interface.